### PR TITLE
Checkout component implementation

### PR DESCRIPTION
**Pull request by DarinaBlazevica**
Checkout Form:
-Add Checkout component
-Email - standard regex for email validation
-Confirm email - check if matches first email entry
-Name/ Surname - regex validation that only letters are typed in && there should be at least two words
-Validation message
-If information is incorrect or is not provided "Order" button is disabled

Order Form: 
-Add Order confirmation form
-Add styling
-Routing to main page when Homepage button is clicked
-Clear cart if Homepage button is clicked

### CODE REVIEW

File: src/app/components/checkout/checkout.component.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+            <label class="checkout_container__user_information--email">email:</label>
+            <input type="email" name="user_email" formControlName="user_email">
+        </div>
+        <div class="error"
+            *ngIf="userData['user_email'].invalid && (userData['user_email'].dirty || userData['user_email'].touched)">
+            <div *ngIf="userData['user_email'].errors?.['required']">Email required</div>
+            <div *ngIf="userData['user_email'].errors?.['pattern']">Provide valid email address</div>
+        </div>
+    </form>
+    <form [formGroup]="checkoutOrderForm">
```
Comment (EgijaLocmele) [inline src/app/components/checkout/checkout.component.html]: Why are you creating form for each of the inputs? Better would be to just have one form tag and include all inside that. Or maybe this way there are some advantages?

Comment (DarinaBlazevica) [reply src/app/components/checkout/checkout.component.html]: Changed

**Review by EgijaLocmele — APPROVED** [review]

File: src/app/components/checkout/checkout.component.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          Validators.pattern('^[a-zA-Z]{2,40}(?: +[a-zA-Z]{2,40})+$'),
+        ]),
+      },
+      {
+        validators: this.EmailMatch('user_email', 'confirm_email'),
+      }
+    );
+  }
+
+  ngOnInit(): void {}
```
Comment (Jalal999) [inline src/app/components/checkout/checkout.component.ts]: if not used, remove

**Review by Jalal999 — APPROVED** [review]

File: src/app/components/checkout/checkout.component.ts (hunk 2 above)
Comment (DarinaBlazevica) [reply src/app/components/checkout/checkout.component.ts]: changed

**Pull request merged**